include/notification_queue.h

```diff
@@ -0,0 +1,66 @@
+#ifndef _NOTIFICATION_QUEUE_H
+#define _NOTIFICATION_QUEUE_H
+/*
+ * FogLAMP notification queue manager.
+ *
+ * Copyright (c) 2018 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Massimiliano Pinto
+ */
+
+#include <logger.h>
+#include <queue>
+#include <condition_variable>
+#include <rule_plugin.h>
+#include <delivery_plugin.h>
+#include <reading_set.h>
+
+class NotificationQueueElement
+{
+	public:
+		NotificationQueueElement(const string& assetName,
+					 ReadingSet* data);
+
+		~NotificationQueueElement();
+
+		const std::string&	getAssetName() { return m_assetName; };
+		ReadingSet*		getAssetData() { return m_readings; };
+
+	private:
+		std::string		m_assetName;
+		ReadingSet*		m_readings;
+};
+
+/**
+ * The NotificationQueue class.
+ */
+class NotificationQueue
+{
+	public:
+		NotificationQueue(const std::string& serviceName);
```

Comment:
If it is a singleton we normally make the constructor private

Reply:
Yes. Other classes have similar public method.

I'll review this in next commits